securedrop/tests/functional/test_source_session_timeout.py

```diff
@@ -1,17 +1,176 @@
-from . import source_navigation_steps
-from . import functional_test
+import socket
+import time
+from dataclasses import dataclass
+from multiprocessing import Process
+from pathlib import Path
 
+from selenium.webdriver.firefox.webdriver import WebDriver
 
-class TestSourceSessions(
-        functional_test.FunctionalTest,
-        source_navigation_steps.SourceNavigationStepsMixin):
+import pytest
+import requests
 
-    session_expiration = 5
+from models import Journalist
+from tests.functional.app_navigators import SourceAppNagivator, WebDriverTypeEnum, get_web_driver
+from tests.functional.db_session import get_database_session
+from tests.functional.factories import SecureDropConfigFactory
+from tests.functional.sd_config_v2 import SecureDropConfig
 
-    def test_source_session_timeout(self):
-        self._source_visits_source_homepage()
-        self._source_clicks_submit_documents_on_homepage()
-        self._source_continues_to_submit_page()
-        self._source_waits_for_session_to_timeout()
-        self.driver.refresh()
-        self._source_sees_session_timeout_message()
+
```

Comment:
What follows is very similar to the existing sd_servers fixture, but with some fixes and tweaks.